diag_manager/diag_util.F90

```diff
@@ -2170,12 +2170,19 @@ CHARACTER(len=128) FUNCTION get_time_string(filename, current_time)
     CHARACTER(len=20) :: hr !< string of current time (output)
     CHARACTER(len=20) :: mi !< string of current time (output)
     CHARACTER(len=20) :: sc !< string of current time (output)
-    CHARACTER(len=128) :: filetail
-
+    CHARACTER(len=MAX_NAME_LENGTH) :: filetail
+    character(len=2) :: first_fms_percent
+    
     format = '("-",i*.*)'
     CALL get_date(current_time, yr1, mo1, dy1, hr1, mi1, sc1)
     len = LEN_TRIM(filename)
-    first_percent = INDEX(filename, '%')
+    do i=1,9
```

Comment:
In this loop instead of just looking for a % sign we look for %\d where \d is a digit 1-9.  This allows other uses of the % sign in the filename.